primitives/src/lib.rs

```diff
@@ -133,6 +141,7 @@ sp_api::decl_runtime_apis! {
         fn authority_data() -> SessionAuthorityData;
         fn session_period() -> u32;
         fn millisecs_per_block() -> u64;
+        fn aleph_bft_version() -> Result<Version, ApiError>;
```

Comment:
In the current solution, it is possible that no version has ever been set and this is the case when the call can fail. This is only relevant at the very beginning when we start to use a version marker. Once the first version change is scheduled but has not yet come into effect, there is this possibility that no `AlephBFTVersion` has been manually set. After the first version change, this is not binding.

Reply:
I would say it's much cleaner to output the default `0` here (the version we start from), instead of an error. Because otherwise what would the first version change look like? The version will be `Err` till some point and then jump to `0` (or rather `1`)? As mentioned somewhere else, you will need to deal with errors that the runtime API is not available at some blocks yet, having internal errors on top of that will only complicate things.

And in case you want to stick to this version then make the error name actually give some indication of what happened.

Reply:
Since the error here pertains to a very slim possibility, let's go with the default-based version. I have introduced changes to reflect this.